--- a/extract.py
+++ b/extract.py
@@ -376,11 +376,11 @@
 # as well as U+3000 is IDEOGRAPHIC SPACE for bug 19052
 EXT_LINK_URL_CLASS = r'[^][<>"\\x00-\\x20\\x7F\\s]'
 ExtLinkBracketedRegex = re.compile(
-    '\\[(((?i)' + '|'.join(wgUrlProtocols) + ')' + EXT_LINK_URL_CLASS + r'+)\\s*([^\\]\\x00-\\x08\\x0a-\\x1F]*?)\\]',
+    '(?i)\\[((' + '|'.join(wgUrlProtocols) + ')' + EXT_LINK_URL_CLASS + r'+)\\s*([^\\]\\x00-\\x08\\x0a-\\x1F]*?)\\]',
     re.S | re.U)
 EXT_IMAGE_REGEX = re.compile(
-    r"""^(http://|https://)([^][<>"\\x00-\\x20\\x7F\\s]+)
-    /([A-Za-z0-9_.,~%\\-+&;#*?!=()@\\x80-\\xFF]+)\\.((?i)gif|png|jpg|jpeg)$""",
+    r"""(?i)^(http://|https://)([^][<>"\\x00-\\x20\\x7F\\s]+)
+    /([A-Za-z0-9_.,~%\\-+&;#*?!=()@\\x80-\\xFF]+)\\.(gif|png|jpg|jpeg)$""",
     re.X | re.S | re.U)
 
 
